lib/file_utils_ext.rb

```diff
@@ -0,0 +1,7 @@
+require 'fileutils'
+
+module FileUtils
+  def self.readable_by_all?(path)
```

Comment:
1. The object under consideration is a `String`, which seems like a class we don't want to be adding methods to. Unless you're suggesting assigning `readable_by_all?` as a singleton method on any given `String` that could end up being used as a path inside `IncludePathsBuilder::Directory`, but that seems brittle in practice.

2. I could use `Pathname` instead of `String`, which was in Bryan's initial code sample. Generally I found that the switching back and forth between the two caused a lot of bookkeeping overhead. For example, you're dealing with `git ls-files` output, should you convert them all into `Pathname` before moving on?

3. I also tried making this a method on `IncludePathsBuilder`, but hit problems due to testing. In a lot of the tests in this repo, we use `Dir.mktmpdir`, which creates a temporary directory. Only problem is, it sets permissions of that temp directory to be `0700`, which caused erroneous bugs when testing code that's so dependent on file and directory permissions. I wanted to centralize this logic in a method that was stubbable, and I didn't want that method to live on `IncludePathsBuilder` since I disliked the idea of stubbing a method on the class under test.